### ISSUE TITLE
Assets forceCopy doesn`t works on common

### ISSUE THREAD

**Issue by DrMabuse23** [yiisoft/yii2#3239]
Hi there i build a module , using advance app, in my Module are some AssetBundle still works fine the issue is that the forcecopy doesnt works right.

my Asset Bundle

````
namespace common\modules\mediating\assets;

use yii\web\AssetBundle;
use backend\assets\AppAsset;

class MediatingAsset extends AssetBundle
{

    public $sourcePath = '@common/modules/mediating/web/';
    public $css = [
        'dist/css/animate.css/animate.css'
    ];
    public $js = [
        "dev/js/model/SortThumbs.js"

    ];

    public $depends = [
        'yii\web\JqueryAsset',
        'yii\web\YiiAsset'
    ];

    /**
     * @var array
     */
    public $publishOptions = [
        'forceCopy' => true
    ];
} 

````
if i change something in my SortThumb.js i always have to clear the asset folder

**Comment by BornToDrink** [comment]
I have the same behavior and cannot understand why it happens.
I have many assets classes, but only 2 of them have "forceCopy => true".
However, only one of them is being refreshed every time, while the other is not.
Double checked everything (also browser cache, but it's not the problem).

assetManager configuration is pretty much empty:

```javascript
'assetManager' => [
	'bundles' => require( __DIR__ . '/../bundle/bundle-development.php' ) ,
],
```

where bundle-development.php is an empty array.

Both asset classes are identical as config (forceCopy => true) only the files are different.
Any ideas?
I also did some debug and the forceCopy seems to be sent correctly, however when the script reach this part here, the situation is different:
http://prntscr.com/8tcxhw

Check mate...

**Comment by samdark** [comment]
@BornToDrink would you please submit it as a separate issue with the full set of things to reproduce it?

**Comment by BornToDrink** [comment]
@samdark - false alarm!
After only 1 hour of debug :D , I found that one of the assets classes had wrongly set $sourcePath the same as the problematic class.
From what I saw during debug, the source path is used as index of bundles array collection.
So in this case the forceCopy was overwritten by a class with the same source path.
And pretty much my case is closed after I fix my path.

However:
As I can see, we cannot use 2 different assets classes with the same $sourcePath, but different settings and files registered.
Is this a possible issue or by design?
